### PR TITLE
Fix observe issue on worker thread

### PR DESCRIPTION
**Pull request by shayanpourvatan**

### CODE REVIEW

File: auth/src/main/java/com/farsitel/bazaar/thread/MainThread.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.farsitel.bazaar.thread
+
+import android.os.Handler
+import android.os.Looper
+
+class MainThread : Handler(Looper.getMainLooper())
```
Comment (PHELAT) [inline auth/src/main/java/com/farsitel/bazaar/thread/MainThread.kt:6]: Make this an `internal class`.

File: auth/src/main/java/com/farsitel/bazaar/thread/ThreadHelper.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.farsitel.bazaar.thread
+
+import android.os.Looper
+
+object ThreadHelper {
```
Comment (PHELAT) [inline auth/src/main/java/com/farsitel/bazaar/thread/ThreadHelper.kt:5]: Make this an `internal object`.

File: sample/src/main/java/com/farsitel/bazaar/sample/auth/MainActivity.kt [hunk 3]
```diff
@@ -83,6 +84,20 @@ class MainActivity : AppCompatActivity() {
...
+    private fun checkInBackThread() {
+
+        Thread {
+            val response = BazaarSignIn.getLastSignedInAccountSync(
+                context = this,
+                owner = this
+            )
+
+            if (response?.isSuccessful == true) {
+                println("Use already login ${response.data?.accountId}")
```
Comment (PHELAT) [inline sample/src/main/java/com/farsitel/bazaar/sample/auth/MainActivity.kt:96]: "User already logged in".

**Review by PHELAT — CHANGES_REQUESTED** [review]

File: auth/src/main/java/com/farsitel/bazaar/thread/ThreadHelper.kt (hunk 2 above)
Comment (shayanpourvatan) [reply auth/src/main/java/com/farsitel/bazaar/thread/ThreadHelper.kt:5]: Done

File: sample/src/main/java/com/farsitel/bazaar/sample/auth/MainActivity.kt (hunk 3 above)
Comment (shayanpourvatan) [reply sample/src/main/java/com/farsitel/bazaar/sample/auth/MainActivity.kt]: Done

File: auth/src/main/java/com/farsitel/bazaar/thread/MainThread.kt (hunk 1 above)
Comment (shayanpourvatan) [reply auth/src/main/java/com/farsitel/bazaar/thread/MainThread.kt]: Done

**Review by PHELAT — APPROVED** [review]
LGTM.

**Pull request merged**